src/main/java/org/opensearch/search/relevance/actionfilter/SearchActionFilter.java

```diff
@@ -143,14 +145,13 @@ private List<ResultTransformerConfiguration> getResultTransformerConfigurations(
 
     // Fetch all index settings for this plugin
     String[] settingNames = supportedResultTransformers.values()
-        .stream()
-        .map(t -> t.getTransformerSettings()
             .stream()
-            .map(s -> s.getKey())
-            .collect(Collectors.toList()))
-        .flatMap(Collection::stream)
-        .collect(Collectors.toList())
-        .toArray(new String[0]);
+            .map(t -> t.getTransformerSettings()
+                    .stream()
+                    .map(Setting::getKey)
+                    .collect(Collectors.toList()))
```

Comment:
Why this change? Is it because collect was being called twice unnecessarily?

Reply:
Yeah... IntelliJ flagged it as a warning, effectively saying "Why are you taking two steps to convert this stream to an array when you can just do it in one step?" With a quick Alt-Enter, it fixed it for me.